colorizer_data/utils.py

```diff
@@ -245,6 +259,60 @@ def update_collection(
         json.dump(collection, f)
 
 
+def write_parquet_or_json_data(
+    data: np.ndarray,
+    outpath: str,
+    filename: str,
+    *,
+    min: Union[float, int, None] = None,
+    max: Union[float, int, None] = None,
+    write_json: bool = False,
+    parquet_compression: str = "brotli",
+    parquet_use_dict: bool = False,
+) -> str:
+    """
+    Writes a numpy array to a JSON or Parquet file, returning the filename of the written file.
+
+    Args:
+        data (`np.ndarray[int | float]`): The numpy array to write.
+        outpath (`str`): The directory to write the file to.
+        filename (`str`): The base filename to write to. The resulting file will be named `{filename}.parquet`
+            or `{filename}.json`.
+
+        min (`int | float | None`): The minimum value of the data array. Written only to JSON files. Defaults to None.
+        max (`int | float | None`): The maximum value of the data array. Written only to JSON files. Defaults to None.
+        write_json (`bool`): If True, writes the data as a JSON file instead of a Parquet file. False by default.
+        parquet_compression (`str`): The compression algorithm to use for Parquet files. Defaults to 'brotli'.
+            See https://arrow.apache.org/docs/python/parquet.html#compression-encoding-and-file-compatibility for more details.
+        parquet_use_dict (`bool`): If True, uses dictionary encoding for parquet files; useful for large dtypes (like strings)
+            with repeated values. Defaults to False.
+
+    Returns:
+        The `str` filename of the written file, ending in either `{filename}.parquet` or `{filename}.json`.
+    """
+    if write_json:
+        data_json = {"data": data.tolist(), "min": min, "max": max}
+        filename = "{}.json".format(filename)
+        with open(outpath + "/" + filename, "w") as f:
+            json.dump(data_json, f)
+        return filename
+    else:
+        df = pd.DataFrame({"data": data})
+        data_arrow = pa.Table.from_pandas(df, preserve_index=False)
+        filename = "{}.parquet".format(filename)
+
+        # In testing, brotli compression + no dictionary encoding gave the smallest overall size
+        # (followed by GZIP + no dict). Because our data is largely numeric and not string-based,
+        # dictionary encoding can double the file size for some float features with highly unique values.
```

Comment:
https://docs.google.com/spreadsheets/d/1g-dnGirInlLHwjvV8lMzqarLf_6a2W0W3gb8bWS5UD8/edit?usp=sharing

Spreadsheet comparing dictionary vs. no dictionary if you want to look!

Reply:
Nit: I would consider just hardcoding the compression and use_dict options until we hear from a user for whom it really makes a big difference.

Reply:
I think I'll keep it as-is because it makes the default compression and dictionary option self-documenting? I can imagine a future where I'd want to play with this and the ability to configure it would be nice